src/parser.cc

```diff
@@ -195,44 +199,82 @@ std::vector<std::pair<std::string, bool>> tokenize_line(const std::string& line)
 
 	return result;
 } // tokenize_lin(string) }}}
+
+std::vector<std::pair<std::string, bool>> split_tokens(std::vector<std::pair<std::string, bool>> tokens)
+{ // {{{
+    std::vector<std::pair<std::string, bool>> result;
+    for (const auto& token : tokens) {
+        if (token.second) { // is quoted?
+            result.push_back(token);
+            continue;
+        }
+
+        const std::string& token_string = token.first;
+        size_t last_operator = 0;
+        for (size_t i = 0; i < token_string.size(); ++i) {
+            if (is_logical_operator(token_string[i])) {
+                const std::string token_candidate = token_string.substr(last_operator, i - last_operator);
+
+                // there is token before logical operator (this is case of binary operators, e.g., a&b)
+                if (!token_candidate.empty())
+                    result.push_back(std::pair<std::string, bool>(
+                            token_candidate, false));
+                result.push_back(std::pair<std::string, bool>(
+                        std::string(1,token_string[i]), false));
+                last_operator = i+1;
+            }
+        }
+
+        if (last_operator == 0) {
+            result.push_back(token);
+        } else if (last_operator != token_string.length()){ // operator was not last, we need parse rest of token
```

Comment:
Extracting call to `lenght` might be a slight optimization, since it is unnecessarily called twice

Reply:
extracted, you little perfnerd.